rest/route/distro.go

```diff
@@ -240,3 +289,41 @@ func (dgh *distroGetHandler) Run(ctx context.Context) gimlet.Responder {
 
 	return resp
 }
+
+////////////////////////////////////////////////////////////////////////
+
+func validateToService(ctx context.Context, apiDistro *model.APIDistro, resourceId string, isNewDistro bool) (*distro.Distro, gimlet.Responder) {
+	i, err := apiDistro.ToService()
+	if err != nil {
+		return nil, gimlet.MakeJSONInternalErrorResponder(errors.Wrap(err, "API Error converting from model.APIDistro to distro.Distro"))
+	}
+	d := i.(*distro.Distro)
+
+	id := model.FromAPIString(apiDistro.Name)
+	if resourceId != id {
+		return nil, gimlet.MakeJSONErrorResponder(gimlet.ErrorResponse{
+			StatusCode: http.StatusBadRequest,
+			Message:    fmt.Sprintf("Distro name is immutable; cannot rename distro resource '%s'", resourceId),
+		})
+	}
+
+	vErrors, err := validator.CheckDistro(ctx, d, &evergreen.Settings{}, isNewDistro)
```

Comment:
I'm a bit curious about what it means to pass an empty settings value. Why do we need the settings value? We could get a populated one.